cucumber/authenticators/features/support/authn_oidc_helper.rb

```diff
@@ -104,6 +127,21 @@ def parse_oidc_id_token
   rescue Exception => err
     raise "Failed to fetch id_token from HTTP response: #{@response_body} with Reason: #{err}"
   end
+
+  def calculate_performance_results(results)
+    @performance_result = {
+        max: results.first.real,
+        avg: 0
+    }
+
+    results.each do |res|
+      real_time = res.real
+      @performance_result[:max] = real_time unless @performance_result[:max] > real_time
+      @performance_result[:avg] += real_time
+    end
+
+    @performance_result[:avg] /= results.length
+  end
```

Comment:
So I actually think this would clean up nicely, and have a more ruby-esque spirit, if we separated the two calculations, rather than trying to combine inside one loop.  The performance difference will be negligible, so we should focus on readability.  And if you think about it, in the above we are re-implementing the functionality provided by `Enumerable.max`.

In fact, the calculation of `max` and `avg` are less than 1-liners.  

And we can actually remove the above method and `ensure_performance_results` altogether.

Instead, at end of the method currently named `run_performance_with_oidc` (which I suggested renaming to `measure_oidc_performance`), rather than calling `calculate_performance_results(results)`, we'll simply save the results:
```ruby
@oidc_perf_results = results.map(&:real)
```

And then we'll update the previous step definition as follows:

```ruby
Then(/^The "{string}" response time should be less than "{float}" seconds$/) do |type_str, threshold|
  type = type_str.downcase.to_sym
  raise "Unexpected Type" unless [:max, :avg].include?(type)
  results = @oidc_perf_results
  actual_time = (type == :avg) ? results.sum.fdiv(results.size) : results.max
  expect(actual_time).to be < threshold
end
```
Allowing us to delete two methods.